src/parser/parser_utils.js

```diff
@@ -1,183 +1,254 @@
-import { Util } from '../util/util';
+import { util } from '../util/util';
 
 /**
- * This class provides support methods to the parsing classes.
- * @export
- * @class ParserUtils
+ * This module provides support methods to the parsing classes.
  */
-export class ParserUtils {
-  /**
-   * Creates an instance of ParserUtils.
-   */
-  constructor() {
-    this.util = new Util();
-  }
 
-  /**
-   * Returns the first element of the given node which nodeName matches the given name.
-   * @param  {Object} node - The node to use to find a match.
-   * @param  {String} name - The name to look for.
-   * @return {Object}
-   */
-  childByName(node, name) {
-    const childNodes = node.childNodes;
+/**
+ * Returns the first element of the given node which nodeName matches the given name.
+ * @param  {Object} node - The node to use to find a match.
+ * @param  {String} name - The name to look for.
+ * @return {Object}
+ */
+function childByName(node, name) {
+  const childNodes = node.childNodes;
 
-    for (let childKey in childNodes) {
-      const child = childNodes[childKey];
+  for (let childKey in childNodes) {
```

Comment:
Right, `for ... in` is not the most suitable option for a `NodeList`, but `for ... of` is completely unsupported in IE (see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...of#Browser_compatibility) and this requires a quite bad polyfill performance-wise to work (see https://daverupert.com/2017/10/for-of-loops-are-bad/).

I wanted to use the quite new `NodeList.prototype.forEach()`, but still it's not supported in old browsers ([and not at all in IE](https://developer.mozilla.org/en-US/docs/Web/API/NodeList/forEach#Browser_Compatibility)). Transpiling it is less expansive than a `for ... of` loop, but still it will increase our bundle size. 

I know the `for ... in` syntax is not cool, but in this case is probably the most performance oriented one. What do you think @kobawan ? :)

Reply:
ah I see, internet explorer again... `for in` seems to be the most performant solution then...